### PR TITLE
ESlint, prettier and editor config setup + code formatting

### PR DESCRIPTION
**Pull request by amanrk28**
## ChangeLog
- Installed new plugins for linting.
- Added eslint + prettier rules and VS Code editor rules for best coding practices.
- Auto formatted the code to implement above rules.

### CODE REVIEW

**Comment by harisarang** [comment]
Hey @amanrk28 I have already installed eslint and prettier in the project just now can you pull it from master.
I would highly suggest add `.vscode` into `.gitignore` because genrally editor configuration are not added in the code base.

Cheers 🥂

**Review by harisarang — CHANGES_REQUESTED** [review]
Hey @amanrk28 I have already installed eslint and prettier in the project just now can you pull it from master.
I would highly suggest add .vscode into .gitignore because genrally editor configuration are not added in the code base.

Cheers 🥂

**Comment by amanrk28** [comment]
> Hey @amanrk28 I have already installed eslint and prettier in the project just now can you pull it from master.
> I would highly suggest add `.vscode` into `.gitignore` because genrally editor configuration are not added in the code base.
> 
> Cheers

Thanks for the heads up. I forgot to add it in .gitignore. All checks have passed. You can review the PR and merge it

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (harisarang) [inline .editorconfig:1]: Even this is related to your editor. So highly feel even this should be added to `.gitignore`

File: .eslintrc.js [hunk 2]
```diff
@@ -1,15 +1,141 @@
+// turn off the rule completely
```
Comment (harisarang) [inline .eslintrc.js:1]: I have added a `.eslintrc.js` which does not break the existing code in repo. If this settings comes to master there could be many merge conflicts for the existing branches. So, I highly recommend sticking with the exisisting one

File: .prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
```
Comment (harisarang) [inline .prettierrc.js:1]: Prettier is already added as plugin with eslint. So, prettier setup won't be necessary.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (harisarang) [inline tsconfig.json:1]: We are not going to use Typescript to this project. So we need not configure this

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+node_modelues/
```
Comment (harisarang) [inline .eslintignore:1]: `node_modules` is by default ignored by eslint and the other files only generated during the CI build. So I feel that this file may not be of use

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+dist/
+build/
+.cache/
+public
```
Comment (harisarang) [inline .eslintignore:5]: `node_modules` is by default ignored by eslint and the other files only generated during the CI build. So I feel that this file may not be of use

**Review by harisarang — CHANGES_REQUESTED** [review]
There are many breaking changes from the existing code because of a different eslint configuration. So I highly recommend to stick with the existing rules. So, we can have the merge conflicts minimal to the possible.

P.S. I feel that you can checkout a new branch from master so that you can jump to the actual development of the project with the existing linting rules

**Pull request closed without merging**